src/web/src/netcfg.ecpp

```diff
@@ -74,20 +75,30 @@ if(!nil.empty()) {
 // Are we asked to just list possible configurations?
 static cxxtools::Regex rex_ncfg("^.*/netcfgs$");
 if(rex_ncfg.match(request.getUrl())) {
-    std::string in;
-    in = augeas->get_cmd_out("match /files/etc/network/interfaces/iface[*]",
-							 true, "\", \"",
+    std::string all_interfaces_str = augeas->get_cmd_out("match /files/etc/network/interfaces/iface[*]",
+							 true, ",",
 							 [](const std::string iface) -> bool {
 							 	return iface == "lo";
 							 }
 							);
-    if(in.empty()) {
+    if(all_interfaces_str.empty()) {
         std::string err =  TRANSLATE_ME ("No configurable interfaces found");
         http_die ("internal-error", err.c_str ());
     }
 
+    std::set<std::string> active_interfaces = get_ifaces ();
+    std::vector<std::string> all_interfaces;
+    cxxtools::split(",", all_interfaces_str, std::back_inserter(all_interfaces));
+
+    std::string out;
+    for (std::string iface : all_interfaces) {
+        if (active_interfaces.find (iface) != active_interfaces.end ()) {
+            out += "\"" + iface + "\",";
+        }
+    }
+    out [out.size() - 1] = ' ';
```

Comment:
what if out.size() = 0?
In fact, this is to erase the comma added previously. Code at line 96 shall be:.
if (!out.empty()) { out += ","; }
out += "\"" + iface + "\""; 
I am wrong?

Reply:
Your point is valid but cosmetical, since we are never going to have a system without any network interfaces.

Reply:
Depends on the consistency of the active_interfaces string. What if get_ifaces() is buggy, returning an empty string for example... My opinion is to support the worst of cases.
It's up to you  ;-)